src/main/java/uk/gov/hmcts/reform/iacaseapi/domain/handlers/postsubmit/AutomaticEndAppealForNonPaymentEaHuTrigger.java

```diff
@@ -52,7 +52,44 @@ public boolean canHandle(
 
         return callback.getEvent() == Event.SUBMIT_APPEAL
                && (remissionType.isPresent() && remissionType.get() == RemissionType.NO_REMISSION)
-               && (appealType.isPresent() && (appealType.get() == AppealType.EA || appealType.get() == AppealType.HU));
+               && (appealType.isPresent() && (appealType.get() == AppealType.EA || appealType.get() == AppealType.HU))
+                && checkScenario(asylumCase);
+    }
+
+    private boolean checkScenario(AsylumCase asylumCase) {
+        Optional<YesOrNo> isDetained = Optional.of(asylumCase.read(APPELLANT_IN_DETENTION, YesOrNo.class).orElse(YesOrNo.NO));
+        Optional<YesOrNo> isAcceleratedDetainedAppeal = Optional.of(asylumCase.read(IS_ACCELERATED_DETAINED_APPEAL, YesOrNo.class).orElse(YesOrNo.NO));
```

Comment:
You can use HandlerUtils class to check if ADA or AgeAssessment. Also add a method there to check if Detained and use the same.

Reply:
Done.